test/specs/jsCodegen/astRejection.tests.ts

```diff
@@ -0,0 +1,191 @@
+			evaluateXPathToFirstNode('/xml[following::element(xml)]', document, null, null, {
+				backend: 'js-codegen',
+			});
+		}, 'Unsupported');
+	});
+	it('rejects unsupported reverse axes', () => {
+		chai.assert.throws(() => {
+			evaluateXPathToFirstNode('/xml[ancestor::element(xml)]', document, null, null, {
+				backend: 'js-codegen',
+			});
+		}, 'Unsupported');
+		chai.assert.throws(() => {
+			evaluateXPathToFirstNode(
+				'/xml[preceding-sibling::element(xml)]',
+				document,
+				null,
+				null,
+				{
+					backend: 'js-codegen',
+				}
+			);
+		}, 'Unsupported');
+		chai.assert.throws(() => {
+			evaluateXPathToFirstNode('/xml[preceding::element(xml)]', document, null, null, {
+				backend: 'js-codegen',
+			});
+		}, 'Unsupported');
+		chai.assert.throws(() => {
+			evaluateXPathToFirstNode('/xml[ancestor-or-self::element(xml)]', document, null, null, {
+				backend: 'js-codegen',
+			});
+		}, 'Unsupported');
+	});
+	it('rejects wildcard with uri', () => {
+		chai.assert.throws(
+			() =>
+				evaluateXPathToFirstNode('/somenamespace:*', document, null, null, {
+					backend: 'js-codegen',
+				}),
+			'Unsupported'
+		);
+	});
+	it('rejects unsupported tests', () => {
+		chai.assert.throws(
+			() =>
+				evaluateXPathToFirstNode(
+					'/xml[self::namespace-node() or self::processing-instruction()]',
+					document,
+					null,
+					null,
+					{
+						backend: 'js-codegen',
+					}
+				),
+			'Unsupported'
+		);
+	});
+	it('rejects unsupported tests combined with the "or" operator', () => {
+		chai.assert.throws(
+			() =>
+				evaluateXPathToFirstNode(
+					'/xml[self::element(tips) or self::processing-instruction()]',
+					document,
+					null,
+					null,
+					{
+						backend: 'js-codegen',
+					}
+				),
+			'Unsupported'
+		);
+	});
+	it('rejects unsupported tests combined with the "and" operator', () => {
+		chai.assert.throws(
+			() =>
+				evaluateXPathToFirstNode(
+					'/xml[self::element(xml) and self::processing-instruction()]',
+					document,
+					null,
+					null,
+					{
+						backend: 'js-codegen',
+					}
+				),
+			'Unsupported'
+		);
+	});
+	it('rejects unsupported base expression (functionCallExpr)', () => {
+		chai.assert.throws(
+			() =>
+				evaluateXPathToFirstNode(
+					'boolean(/xml) and self::element(xml)',
+					document,
+					null,
+					null,
+					{
+						backend: 'js-codegen',
+					}
+				),
+			'Unsupported'
+		);
+	});
+	it('rejects name tests with namespace URI', () => {
+		chai.assert.throws(() => {
+			evaluateXPathToFirstNode('/Q{https://example.com}xml', document, null, null, {
+				backend: 'js-codegen',
+			});
+		}, 'Unsupported');
+	});
+	it('rejects library modules', () => {
+		chai.assert.throws(() => {
+			evaluateXPathToFirstNode(
+				`
+module namespace test = "http://www.example.org/mainmodules.tests#1";
+
+declare %public function test:hello($a) {
+"Hello " || $a
+};
+`,
+				null,
+				null,
+				null,
+				{ backend: 'js-codegen' }
+			);
```

Comment:
I like to add explicit checks for the correct error type so that we can be sure we are not throwing a TypeError by accident because we made a typo in this error handling code